In [1]:
import pandas as pd
df= pd.read_csv("allbacks.csv")

In [2]:
print(df.shape)
display(df.head())
print(df.dtypes)

(15, 5)


,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb


rownames    int64
volume      int64
area        int64
weight      int64
cover         str
dtype: object


In [12]:
display(df[["cover"]])

,cover
0,hb
1,hb
2,hb
3,hb
4,hb
5,hb
6,hb
7,pb
8,pb
9,pb


In [3]:
df["cover"].value_counts()

cover
pb    8
hb    7
Name: count, dtype: int64

The median cover would be paperback. When the books are put in order like shown in the data, the first 7 are hardback making the 8th, which would be the middle, a paperback.
Written out, it would look like: HB,HB,HB,HB,HB,HB,HB,**PB**,PB,PB,PB,PB,PB,PB,PB

In [4]:
df["weight"].mean()

np.float64(706.6666666666666)

If we add up all the weights it would be 10,600 divided by number of books, which is 15. 
    10,600/15= 706.67
This amount matches the python result of 706.67 (rounded two decimal places)

In [10]:
df[df["cover"] == "hb"]["weight"].mean()

np.float64(796.4285714285714)

In [11]:
df[df["cover"] == "pb"]["weight"].mean()

np.float64(628.125)

Hardback books weigh on average about 168 more than paperback books. Paperback books weighing less can be beneficial for the library to move the books as they weigh less. 

AI USE: Chat gpt
explain value_counts data results

https://github.com/emflores93/cs82a-portfolio